### PR TITLE
Enable `MethodInputParametersMustBeFinal`

### PR DESCRIPTION
**Pull request by gfukushima**
## PR Description
This PR add some missing `final` and enables the Error Prone `MethodInputParametersMustBeFinal` 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/testFixtures/java/tech/pegasys/teku/spec/datastructures/util/MerkleTree.java [hunk 1]
```diff
@@ -113,25 +114,25 @@ public List<Bytes32> getProof(int itemIndex) {
...
   }
 
-  private Bytes32 calcViewBoundaryRoot(int depth, int viewLimit) {
+  private Bytes32 calcViewBoundaryRoot(final int depth, final int viewLimit) {
     if (depth == 0) {
       return zeroHashes.get(0);
     }
-    depth -= 1;
-    Bytes32 deeperRoot = calcViewBoundaryRoot(depth, viewLimit);
+    int mutableDepth = depth - 1;
```
Comment (rolfyone) [inline ethereum/spec/src/testFixtures/java/tech/pegasys/teku/spec/datastructures/util/MerkleTree.java]: wait so this is just off by 1? looks like this is not altered after decrementing.

File: infrastructure/ssz/src/test/java/tech/pegasys/teku/infrastructure/ssz/SszListHintsTest.java [hunk 2]
```diff
@@ -93,18 +97,19 @@ <TElement extends SszData> void assertListElements(
...
 
     listVariants1.forEach(
         listVariant1 ->
-            listVariants2.forEach(
-                listVariant2 -> assertListEquals(type, listVariant1, listVariant2)));
+            listVariants2.forEach(listVariant2 -> assertListEquals(listVariant1, listVariant2)));
   }
 
   <TElement extends SszData> void assertListEquals(
-      SszListSchema<TElement, ?> type, SszList<TElement> list1, SszList<TElement> list2) {
```
Comment (rolfyone) [inline infrastructure/ssz/src/test/java/tech/pegasys/teku/infrastructure/ssz/SszListHintsTest.java:107]: assuming type wasn't referenced

**Review by rolfyone — APPROVED** [review]
minor bits...

File: infrastructure/ssz/src/test/java/tech/pegasys/teku/infrastructure/ssz/SszListHintsTest.java (hunk 2 above)
Comment (gfukushima) [reply infrastructure/ssz/src/test/java/tech/pegasys/teku/infrastructure/ssz/SszListHintsTest.java:107]: it wasn't

File: ethereum/spec/src/testFixtures/java/tech/pegasys/teku/spec/datastructures/util/MerkleTree.java (hunk 1 above)
Comment (gfukushima) [reply ethereum/spec/src/testFixtures/java/tech/pegasys/teku/spec/datastructures/util/MerkleTree.java]: you're right this is just single decrement operation done on top of the input param `depth`. New variable can actually be `final` as well

Comment (rolfyone) [reply ethereum/spec/src/testFixtures/java/tech/pegasys/teku/spec/datastructures/util/MerkleTree.java]: probably worth just looking where this is called too and see whether the depth passed in is just wrong lol

Comment (gfukushima) [reply ethereum/spec/src/testFixtures/java/tech/pegasys/teku/spec/datastructures/util/MerkleTree.java]: Mutation of this index is actually part of the recursive approach used in the this method, so I think we should keep it.

Comment (rolfyone) [reply ethereum/spec/src/testFixtures/java/tech/pegasys/teku/spec/datastructures/util/MerkleTree.java]: i do still think final, and potentially deeperRoot could be final also

so mutableDepth => final internalDepth
and deeperRoot -> final... at least on the code here...

**Pull request merged**